### PR TITLE
WT-12752 Remove LSM documentation

### PR DESCRIPTION
**Pull request by lukech**
This PR aims to remove all LSM-related documentation as part of PM-3441, after LSM tests and feature code being removed in WT-12750 and WT-12751.

### CODE REVIEW

File: src/docs/arch-glossary.dox [hunk 1]
```diff
@@ -22,13 +22,11 @@ they are used in other settings.
 <tr><td>block<td>block manager<td>the smallest granularity of data that is read from or written to a Btree file.  A btree page, when it appears on disk, may consist of one or more blocks.
 <tr><td>block compression<td>btree<td>after pages of btrees are encoded to be written, they may be compressed according to a configured algorithm.  Compressors may be added to WiredTiger via the extension mechanism.
 <tr><td>block manager<td>block manager<td>an internal API that abstracts writing, reading and allocating blocks to and from files.  A block manager is associated with a Btree.
-<tr><td>Bloom filter<td>lsm<td>a data structure that identifies that a key cannot be present.  Used by LSM.
```
Comment (luke-pearson) [inline src/docs/arch-glossary.dox:25]: We still support bloom filters so I think we only need to drop "Used by LSM." here.

File: src/docs/arch-glossary.dox [hunk 2]
```diff
@@ -65,8 +63,6 @@ they are used in other settings.
 <tr><td>key prefix compression<td>btree<td>a technique to store identical key prefixes only once per page.  See @ref file_formats_compression
 <tr><td>kv or k/v<td>btree<td>a key/value pair in a Btree.
 <tr><td>leaf page<td>btree<td>either in-memory or on disk, a page in a Btree that has a pairs of keys and values, and no pages beneath it.
-<tr><td>log structured merge tree (or LSM tree)<td>lsm<td>a data structure that stores keys and values (as a Btree), but composed of potentially many trees (Btrees in WiredTiger).
```
Comment (luke-pearson) [inline src/docs/arch-glossary.dox:68]: We could in theory keep this as its a definition, but happy to remove it too.

File: src/docs/arch-glossary.dox (hunk 1 above)
Comment (lukech) [reply src/docs/arch-glossary.dox:25]: Okay sure. Added it back in [03bf29c](https://github.com/wiredtiger/wiredtiger/pull/11299/commits/03bf29c20b2d416b3e304477b5cecc0c684ebe9f)

File: src/docs/arch-glossary.dox (hunk 2 above)
Comment (Sean04) [reply src/docs/arch-glossary.dox:68]: I don't think we need to keep it if the documentation has no references to it. The definition will still be present in older versions of the docs which are accessible.

**Review by luke-pearson — APPROVED** [review]
LGTM.

File: src/docs/data-sources.dox [hunk 3]
```diff
@@ -67,9 +64,9 @@ Advanced applications may also open the following low-level cursor types:
 WiredTiger provides access to data from a variety of sources.  At the
 lowest level, data may be stored in a file using a tree structure.  A
 relational schema supporting tables, indices and column groups is
-layered on top of file.  Additional sources include LSM trees and
-statistics, and applications can further extend the supported types by
-implementing the ::WT_DATA_SOURCE interface.
+layered on top of file. Additional sources include statistics.
```
Comment (Sean04) [inline src/docs/data-sources.dox:67]: Does it make sense to day "Additional sources include statistics, log files and backups"?

Comment (lukech) [reply src/docs/data-sources.dox]: Sure, sounds good. See [8f07ddc](https://github.com/wiredtiger/wiredtiger/pull/11299/commits/8f07ddc25363fc67b6183a9f8a60be26a3430309)

**Review by Sean04 — APPROVED** [review]
lgtm

**Pull request merged**